pkg/ccl/backupccl/restore_planning.go

```diff
@@ -1592,14 +1593,39 @@ func doRestorePlan(
 	}()
 
 	currentVersion := p.ExecCfg().Settings.Version.ActiveVersion(ctx)
+
+	// We support restoring a backup that was taken on a cluster that is on a
+	// version >= the earliest binary version that this binary can interoperate with.
+	minimumRestorableVersion := p.ExecCfg().Settings.Version.BinaryMinSupportedVersion()
+	if !build.IsRelease() {
+		if minimumRestorableVersion.Major < clusterversion.DevOffset {
+			return errors.AssertionFailedf("minimum restoreable version %s is less than the dev offset",
+				minimumRestorableVersion)
+		}
+		minimumRestorableVersion.Major -= clusterversion.DevOffset
```

Comment:
could you clarify in a comment why this decrement is necessary? my read of the code is that `BinaryMinSupportedVersion()` always returns `v22_2`, but maybe i misread things?

Reply:
When running in a development branch as defined by this bool - https://github.com/cockroachdb/cockroach/blob/2d633784329ae8a1d14854a827db978296e0ed6c/pkg/clusterversion/cockroach_versions.go#L846, all versions including the `BinaryMinSupportedVersion` are offset by `DevOffset`. If we didn't decrement the `MinimumBinaryVersion` by the dev offset then a backup taken on say 22.2 would not be restoreable into our branch because our `MinimumBinaryVersion` is `1000022.2`. Since 22.2 is far less than this number we would reject the restore. 

This has however got me thinking that the condition we decrement the `BinaryMinSupportedVersion` should not be `IsRelease()` but instead the `developmentBranch` bool linked above.

Reply:
let me know if you want company in the "why do we have _both_ a `IsRelease()` and `developmentBranch`" rabbit hole :)